articles/pointers-in-c/index.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/pointers-in-c/
+title: Pointers in C
+description: Pointers are a hard concept to grasp for a lot of new programmers. This article will help you understand what pointers are and how to use them.   
+author: nimra-aftab
+date: 2020-08-28T00:00:00-12:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/pointers-in-c/hero.jpg
+    alt: c pointers c++ variable
+---
+Pointers are common in the C programming language. Since C doesn't have built-in arrays and strings, programmers use pointers instead. This article will give a basic introduction to pointer variables.
+<!--more-->
+
+### An Introduction to Pointers
+
+Every variable in your program uses up at least one byte in the computer memory. Each byte has an address. *Pointer variables* store the address of a byte in memory. A *pointer* points to something--an `int`, `double`, `char`, etc . To declare a *pointer variable*, we use an asterisk (*). We use ampersand (&) to get the address of the variable being pointed at. 
+``` c
+int i = 5;
+int *p;
+p = &i; /* We say p 'is a pointer to' i */ 
+```
+Notice that we had to declare the variable type of `p` : a pointer to an integer, or type `int *`. 
+A common mistake is to say the pointer variable declared above is `*p`.  The variable above is __not__ declared as `*p`. It is the variable `p` of type `int *`. In fact, `*p` means something else: it says to dereference the pointer `p`. When we __dereference__ a pointer, we access what is being stored at the memory address. Here's a sample program:
+
+```c
+#include <stdio.h> 
+int main(){
+    int i = 5;
+    int *p;
+    p = &i; /* p points to i */
+    printf(" The value of i is %d \n", *p); // Dereferencing p gives i
+    printf(" The memory address of i is %p\n ", p); 
+    return 0;
+}
+``` 
```

Comment:
Can you add the output of this program for those who don't run the code?

Reply:
I've added the program output as per your suggestion on line 42.